src/cdk/v2/destinations/accoil_analytics/procWorkflow.yaml

```
@@ -0,0 +1,122 @@
+bindings:
+  - name: EventType
+    path: ../../../../constants
+  - path: ../../bindings/jsontemplate
+    exportAll: true
+  - name: removeUndefinedAndNullValues
+    path: ../../../../v0/util
+  - name: base64Convertor
+    path: ../../../../v0/util
+  - path: ./utils
+
+steps:
+  - name: validateInput
+    template: |
+      $.assert(.message.type, "message type is not present. Aborting message.");
+      $.assert(.message.type in {{$.EventType.([.TRACK, .PAGE, .SCREEN, .IDENTIFY, .GROUP])}},
+        "message type " + .message.type + " is not supported");
+      $.assertConfig(.destination.Config.apiKey, "apiKey must be supplied in destination config");
+
+#   Note our auth does not require a password in basic auth just the string "key:"
+  - name: prepareContext
+    template: |
+      $.context.messageType = .message.type.toLowerCase();
+      $.context.payload = {
+        "type": $.context.messageType
+      };
+      $.context.finalHeaders = {
+          "Authorization": "Basic " + $.base64Convertor(.destination.Config.apiKey + ":"),
+          "Content-Type": "application/json"
+        };
+      $.context.endpoint = $.endpointUrl(.destination.Config.apiKey);
+
+  - name: trackPayload
+    condition: $.context.messageType == {{$.EventType.TRACK}}
+    template: |
+      $.context.payload.event = .message.event;
+      $.context.payload.userId = .message.().({{{{$.getGenericPaths("userIdOnly")}}}})
+      $.context.payload.timestamp = .message.().({{{{$.getGenericPaths("timestamp")}}}})
+
+  - name: pagePayload
+    condition: $.context.messageType == {{$.EventType.PAGE}}
+    template: |
+      $.context.payload.userId = .message.().({{{{$.getGenericPaths("userIdOnly")}}}});
+      $.context.payload.name = .message.name;
+      $.context.payload.timestamp = .message.().({{{{$.getGenericPaths("timestamp")}}}});
+
+  - name: screenPayload
+    condition: $.context.messageType == {{$.EventType.SCREEN}}
+    template: |
+      $.context.payload.userId = .message.().({{{{$.getGenericPaths("userIdOnly")}}}});
+      $.context.payload.name = .message.name;
+      $.context.payload.timestamp = .message.().({{{{$.getGenericPaths("timestamp")}}}});
+
+  - name: identifyPayload
+    condition: $.context.messageType == {{$.EventType.IDENTIFY}}
+    template: |
+      $.context.payload.userId = .message.().({{{{$.getGenericPaths("userIdOnly")}}}});
+      $.context.payload.traits = .message.traits ?? .message.context.traits;
+      $.context.payload.timestamp = .message.().({{{{$.getGenericPaths("timestamp")}}}});
+
+  - name: groupPayload
+    condition: $.context.messageType == {{$.EventType.GROUP}}
+    template: |
+      $.context.payload.userId = .message.().({{{{$.getGenericPaths("userId")}}}});
+      $.context.payload.groupId = .message.groupId;
+      $.context.payload.timestamp = .message.().({{{{$.getGenericPaths("timestamp")}}}});
+      $.context.payload.traits = .message.traits ?? .message.context.traits;
```

```suggestion
      $.context.payload.traits = .message.traits;
```
This should be sufficient, as we don't get context.traits for group calls.

---

It's not compulsory. Right?

https://www.rudderstack.com/docs/event-spec/standard-events/group/#group-fields

> Identity data (user traits) will be automatically added to group calls from the most recent identify call, so you do not need to add it manually. This information will be included in a traits object in the context fields of the payload. Note that group calls also automatically handle anonymousId values associated with the user.

---

We might get context.traits but that will contain user-related data. But group-related data should be in the root level traits only.

---

got it.